pkg/provisioner/ironic/ironic.go

```diff
@@ -535,6 +535,21 @@ func (p *ironicProvisioner) InspectHardware() (result provisioner.Result, detail
 				err = nil
 				return
 			default:
+				// Update the node settings in ironic to ensure we
+				// have the boot mode configured, at least.
+				var updates nodes.UpdateOpts
+				updates, err = p.getUpdateOptsForNode(ironicNode)
```

Comment:
This would seem a lot lower-risk if we just refactored the code for setting the boot mode option out of getUpdateOptsForNode() and only set that one here.